_posts/2020-03-27-simulating-a-pandemic-with-elixir.md

```diff
@@ -0,0 +1,142 @@
+---
+layout: post
+title: Simulating a pandemic with Elixir
+author: jgke
+excerpt: >
+  Wash your hands! Just how many deaths can you prevent by this simple trick?
+  Take a look with this Elixir-based pandemic simulator.
+tags:
+ - Elixir
+---
+
+<img src="/img/simulating-pandemics-with-elixir/simulation.gif"
+     alt="Simulation animation with 9 nodes" />
+
+As I'm typing this, Finland has entered a state of emergency over COVID-19 and
+it is generally recommended to apply social distancing for the time being. The
+drastic measures aim to lessen the exponential curve and thus the impact of the
+virus on Finland's healthcare services. How much does this actually help? Let's
+find out.
+
+Elixir is a dynamic language designed with scalability in mind. Built around
+message passing and lightweight procesesses, building a simulator with nodes (=
+persons) communicating to each other sounds like a good opportunity to use
+Elixir.
+
+Additionally, we want to visualize the social graphs for persons. GraphViz,
+also known by its markup language dot, is a piece of software designed for
+drawing graphs. It scales nicely into reasonable graph sizes.
+
+What is message passing?
+------------------------
+
+A lot of Elixir is based around passing around messages. If you squint your
+eyes just right, it's actually pretty close to calling methods in some
+object-oriented language.
+
+```elixir
+{:ok, pid} = Agent.start_link(fn -> %{infected: false} end)
+Agent.update(pid, &Map.put(&1, :infected, true))
+true = Agent.get(pid, &Map.get(&1, :infected))
+```
+
+A process is created here using the Agent abstraction, which is a wrapper
+around GenServer. Essentially it creates a separate Erlang process and manages
```

Comment:
What is GenServer? Ah, I think the next sentence explains what it is. Maybe something like "… around GenServer. GenServer is <something> that creates a separate…" would be clearer.